content/docs/practices/remote_write.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Remote Write Tuning
+sort_rank: 8
+---
+
+# Remote write tuning
+
+Prometheus implements sane defaults for remote write, but many users have
+different requirements and would like to optimize their remote settings.
+
+This page describes the tuning parameters available via the [remote write
+configuration.](/docs/prometheus/latest/configuration/configuration/#remote_write)
+
+## Remote write characteristics
+
+Each remote write destination starts a queue which reads from the write-ahead
+log (WAL), writes the samples into a queue owned by a shard, which then sends a
+request to the configured endpoint. The flow of data looks like:
+
+```
+      |-->  queue (shard_1)   --> remote endpoint
+WAL --|-->  queue (shard_...) --> remote endpoint
+      |-->  queue (shard_n)   --> remote endpoint
+```
+
+When one shard backs up and fills its queue, Prometheus will block reading from
+the WAL into any shards. Failures will be retried without loss of data unless
+the remote endpoint remains down for more than 2 hours. After 2 hours, the WAL
+will be compacted and data that has not been sent will be lost.
+
+During operation, Prometheus will continuously calculate the the optimal number
+of shards to use based on the incoming sample rate, number of outstanding
+samples not sent, and time taken to send each sample.
+
+### Memory usage
+
+Using remote write increases the memory footprint of Prometheus. Most users
+report ~25% increased memory usage, but that number is dependent on the shape
+of the data. A mapping of series id to label values is stored for each series
+in the WAL, so large amounts of series churn significantly increases memory
+usage.
```

Comment:
Might be good to clarify how and which portions of the on-disk WAL are mapped / copied into memory, to give more context here.

Reply:
This paragraph tries to specify that it is a mapping of series id to label values, do you have any suggestions for how to clarify that?

Reply:
Ahh, I parenthesized sentence fragments together incorrectly in my mind: that for each series, there is a mapping cached in the WAL, not that for each series in the WAL, there is a mapping cached (in memory).

How about (to make the structure extra clear): "For each series in the WAL, the remote write code caches a mapping of series ID to label values."

Reply:
:+1: That sounds good. I have made the change.